tools/adventure-pack/goodies/java/UnionFind.java

```diff
@@ -0,0 +1,24 @@
+class UnionFind {
```

Comment:
Please also add your name to the contributors list in `package.json` in the adventure pack directory!

Reply:
Done.